### PR TITLE
Selenium initial setup and first frontend test

### PR DESCRIPTION
**Pull request by karthikmsqa**
# Description

This PR includes the frontend automated selenium tests and documentation.

-  Jira  https://justice.gov.bc.ca/jira/browse/FLA-43
- Set up a tests directory
- Add dependencies in a local pom.xml

## Type of change
- [x] New tests(non-breaking change which adds automated tests)
- [x] Documentation

## How Has This Been Tested?

Ran the test on localhost

## Checklist:
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/util/DriverClass.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            switch (browser) {
+                case "chrome":
+                    WebDriverManager.chromedriver().setup();
+                    driver = new ChromeDriver();
+                    break;
+                case "firefox":
+                    WebDriverManager.firefoxdriver().setup();
+                    driver = new FirefoxDriver();
+                    break;
+                case "edge":
```
Comment (akroon3r) [inline tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/util/DriverClass.java:31]: might want to add ie too if possible?

File: tests/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+## Frontend Automation Folder Structure
```
Comment (sdevalapurkar-bcgov) [inline tests/README.md:1]: usually readme starts with name of the directory as a main heading
```
Frontend Tests

Folder structure
```

File: tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/util/ReadConfig.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Properties;
+
+public class ReadConfig {
+
+    Properties prop;
+
+    public ReadConfig() throws IOException {
+            File src = new File("src\\main\\java\\ca\\bc\\gov\\open\\jagefilingapi\\qa\\config\\config.properties");
```
Comment (sdevalapurkar-bcgov) [inline tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/util/ReadConfig.java:15]: is there no way to pass in a relative path?
something like
```
../../config/config.properties
```

File: tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/util/ReadConfig.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+                prop = new Properties();
+                prop.load(fis);
+        }
+
+        public String getBaseUrl() {
+            return prop.getProperty("baseUrl");
+        }
+
+        public String getBrowser() {
+        return prop.getProperty("browser");
```
Comment (sdevalapurkar-bcgov) [inline tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/util/ReadConfig.java:27]: nit: tabbing

File: tests/src/test/java/ca/bc/gov/open/jagefilingapi/qa/tests/LandingPageTest.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Test
+    public void verifyLandingPageTitleTest() {
+        landingPage = new LandingPage(driver);
+
+    String actualTitle = landingPage.verifyLandingPageTitle();
+    String expTitle = "eFiling Frontend";
+
+    Assert.assertEquals(expTitle, actualTitle);
+    log.info("Correct page title is verified.");
```
Comment (sdevalapurkar-bcgov) [inline tests/src/test/java/ca/bc/gov/open/jagefilingapi/qa/tests/LandingPageTest.java:44]: nit: tabbing (although these tabbing issues may just be a github issue so nbd)

File: tests/src/test/resources/log4j.properties [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+# Appender which writes to a file
+log4j.appender.file=org.apache.log4j.RollingFileAppender
+log4j.appender.file.File=application.log
+
+# Defining maximum size of a log file
+log4j.appender.file.MaxFileSize=10mb
+log4j.appender.file.MaxBackupIndex=10
+log4j.appender.file.layout=org.apache.log4j.PatternLayout
+log4j.appender.file.layout.ConversionPattern=%d{ISO8601} %5p [%t] %c{1}:%L - %m%n
+log4j.appender.file.Append=false
```
Comment (sdevalapurkar-bcgov) [inline tests/src/test/resources/log4j.properties:18]: nit: missing newline at end of file

File: tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/util/DriverClass.java (hunk 1 above)
Comment (karthikmsqa) [reply tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/util/DriverClass.java:31]: added ie browser. Thanks!

File: tests/README.md (hunk 2 above)
Comment (karthikmsqa) [reply tests/README.md:1]: Modified the heading. Thanks!

File: tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/util/ReadConfig.java (hunk 3 above)
Comment (karthikmsqa) [reply tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/util/ReadConfig.java:15]: I tried using the suggested relative path prior to using the existing one. Couldn't get to work. Assumption was due to the package name in the path. Will keep trying solutions. Thanks.

File: tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/util/ReadConfig.java (hunk 4 above)
Comment (karthikmsqa) [reply tests/src/main/java/ca/bc/gov/open/jagefilingapi/qa/util/ReadConfig.java:27]: Fixed. Thanks

File: tests/src/test/java/ca/bc/gov/open/jagefilingapi/qa/tests/LandingPageTest.java (hunk 5 above)
Comment (karthikmsqa) [reply tests/src/test/java/ca/bc/gov/open/jagefilingapi/qa/tests/LandingPageTest.java:44]: Fixed tabbing. Thanks!

File: tests/src/test/resources/log4j.properties (hunk 6 above)
Comment (karthikmsqa) [reply tests/src/test/resources/log4j.properties:18]: Added new line. Thanks

File: tests/README.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
+## Frontend Tests 
+
+##Folder Structure
```
Comment (sdevalapurkar-bcgov) [inline tests/README.md:3]: need to add a space here for formatting to work properly

File: tests/README.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
+## Frontend Tests
```
Comment (sdevalapurkar-bcgov) [inline tests/README.md:1]: nit: maybe this should be a larger heading than subheadings?
```
# Frontend Tests
```

or we make the sub headings smaller? what do you think?

File: tests/README.md (hunk 7 above)
Comment (karthikmsqa) [reply tests/README.md]: fixed. Thanks!

File: tests/README.md (hunk 8 above)
Comment (karthikmsqa) [reply tests/README.md:1]: Good idea, Thank you.

**Pull request merged**